website/components/remote-plugin-docs/utils/merge-remote-plugins.js

```diff
@@ -0,0 +1,164 @@
+const path = require('path')
+const fetchGithubFile = require('./fetch-github-file')
```

Comment:
This `merge-remote-plugins` function runs the job of:
1. fetching remote plugin `nav-data` (each remote plugin may contain multiple documented components) and
2. merging each documented component into the `navData` tree.

Note that this "merging" occurs _before_ we validate the `navData`. This means that if the process of merging the remote plugin entries into the `navData` creates an invalid structure (such as duplicate paths), those issues **will** be surfaced as errors.